src/lib.rs

```diff
@@ -425,6 +425,26 @@ impl_trait!(f64);
 impl_trait!(usize);
 impl_trait!(isize);
 
+impl TypeName for () {
+    const TYPE_NAME: &'static str = "()";
+}
+
+unsafe impl ToByteSlice for () {
+    fn to_byte_slice<T: AsRef<[()]> + ?Sized>(slice: &T) -> &[u8] {
+        let slice = slice.as_ref();
+        let len = slice.len() * mem::size_of::<()>();
+        unsafe { slice::from_raw_parts(slice.as_ptr() as *const u8, len) }
```

Comment:
You could do both here by simple returning `&[]` / `&mut []` without any unsafe code.

Reply:
Of course :see_no_evil: I was lazy (and following RustConf) and just took the output of `cargo expand` ...

Reply:
Thanks, no worries :)